File: 617.merge.two.binary.trees.rb
# Definition for a binary tree node.
# class TreeNode
#     attr_accessor :val, :left, :right
#     def initialize(val)
#         @val = val
#         @left, @right = nil, nil
#     end
# end

# @param {TreeNode} t1
# @param {TreeNode} t2
# @return {TreeNode}
def merge_trees(t1, t2)
    if t1 == nil and t2 == nil then return nil end
    val1, val2, l1, l2, r1, r2 = 0, 0, nil, nil, nil, nil
    if t1 != nil
        val1 = t1.val
        l1 = t1.left
        r1 = t1.right
    end
    if t2 != nil
        val2 = t2.val
        l2 = t2.left
        r2 = t2.right
    end
    root = TreeNode.new(val1 + val2)
    root.left = merge_trees(l1, l2)
    root.right = merge_trees(r1, r2)
    return root
end

File: 617. Merge Two Binary Trees.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def mergeTrees(self, root1: Optional[TreeNode], root2: Optional[TreeNode]) -> Optional[TreeNode]:
        l1, l2 = None, None
        r1, r2 = None, None
        v1, v2 = 0, 0
        if root1:
            v1 = root1.val
            l1, r1 = root1.left, root1.right
        if root2:
            v2 = root2.val
            l2, r2 = root2.left, root2.right
        if (not root1) and (not root2):
            return None
        node = TreeNode(v1+v2)
        node.left = self.mergeTrees(l1, l2)
        node.right = self.mergeTrees(r1, r2)
        return node